pkg/auth/iam/iam.go

```diff
@@ -477,3 +500,41 @@ func respondError(httpStatus, errorCode int, errorMessage string) restful.Servic
 		Message: string(messageByte),
 	}
 }
+
+// SetDevStackTracer used for activate verbose error message in non-prod environment
+func SetDevStackTracer() {
+	DevStackTraceable = true
+	realmName, realmNameExists := os.LookupEnv("REALM_NAME")
+	if !realmNameExists {
+		DevStackTraceable = false
+		logrus.Warn("unable to get REALM_NAME environment variable, DevStackTracer disabled")
+	}
+
+	realmLives, realmLiveExists := os.LookupEnv("REALM_LIVE")
+	if !realmLiveExists {
+		realmLives = constant.DisabledDevStackTracerEnv
+	}
+
+	if r := strings.Contains(realmLives, realmName); r {
+		DevStackTraceable = false
+		logrus.Warn("DevStackTracer disabled in Live environment")
+	}
```

Comment:
strictly speaking, this method has some logic problems: 
1. if we set it to false in L509, from your code, it no needs to execute the following code(L513-L521)
2. can we really use strings.Contains(realmLives, realmName) to distinguish the realm?

Reply:
1. noted, yeah we can diretly return once DevStackTraceable already set to false
2. I think yes, the point here is to check rather the realm is Live environment or not. Right?
3. I see... It's not a problem wheter DevStackTraceable is enabled or disabled
@angelo-chan

Reply:
for 2, I do not think so, we should match each env, not contains in a string; take an example, the value is development if the current env is dev, shall we treat it the same
for 3, This does have a problem. For a real product, we should always eliminate the warnings when starting, actually enable/disable is a common behavior and should not be a warning to confuse the user.

Reply:
I see.. I got the point. We should match each because "development" is contains "dev"

yes... I got it too, we should avoid unnecessary warning when starting a service